sgtm/apollo/gtm/variable_template.tpl

```diff
@@ -0,0 +1,257 @@
+  logToConsole('return_value :' + return_value);
+  
+  return return_value;
+});
+
+
+___SERVER_PERMISSIONS___
+
+[
+  {
+    "instance": {
+      "key": {
+        "publicId": "logging",
+        "versionId": "1"
+      },
+      "param": [
+        {
+          "key": "environments",
+          "value": {
+            "type": 1,
+            "string": "debug"
+          }
+        }
+      ]
+    },
+    "clientAnnotations": {
+      "isEditedByUser": true
+    },
+    "isRequired": true
+  },
+  {
+    "instance": {
+      "key": {
+        "publicId": "access_response",
+        "versionId": "1"
+      },
+      "param": [
+        {
+          "key": "writeResponseAccess",
+          "value": {
+            "type": 1,
+            "string": "specific"
+          }
+        },
+        {
+          "key": "writeHeaderAccess",
+          "value": {
+            "type": 1,
+            "string": "any"
+          }
+        },
+        {
+          "key": "writeStatusAllowed",
+          "value": {
+            "type": 8,
+            "boolean": true
+          }
+        },
+        {
+          "key": "writeHeadersAllowed",
+          "value": {
+            "type": 8,
+            "boolean": true
+          }
+        },
+        {
+          "key": "writeBodyAllowed",
+          "value": {
+            "type": 8,
+            "boolean": true
+          }
+        }
+      ]
+    },
+    "clientAnnotations": {
+      "isEditedByUser": true
+    },
+    "isRequired": true
+  },
+  {
+    "instance": {
+      "key": {
+        "publicId": "send_http",
+        "versionId": "1"
+      },
+      "param": [
+        {
+          "key": "allowedUrls",
+          "value": {
+            "type": 1,
+            "string": "specific"
+          }
+        },
+        {
+          "key": "urls",
+          "value": {
+            "type": 2,
+            "listItem": [
+              {
+                "type": 1,
+                "string": "https://sheets.googleapis.com/"
+              }
+            ]
+          }
+        }
+      ]
+    },
+    "clientAnnotations": {
+      "isEditedByUser": true
+    },
+    "isRequired": true
+  },
+  {
+    "instance": {
+      "key": {
+        "publicId": "use_google_credentials",
+        "versionId": "1"
+      },
+      "param": [
+        {
+          "key": "allowedScopes",
+          "value": {
+            "type": 2,
+            "listItem": [
+              {
+                "type": 1,
+                "string": "https://www.googleapis.com/auth/spreadsheets"
+              }
+            ]
+          }
+        }
+      ]
+    },
+    "clientAnnotations": {
+      "isEditedByUser": true
+    },
+    "isRequired": true
+  }
+]
+
+
+___TESTS___
+
+scenarios:
+- name: Test case to show getting data from the spreadsheet behaves as expected
+  code: |-
+    const mockData = {
+      // Mocked field values
+      spreadsheet_id: '1XCntdqorwXKCvFeE2bh51OJB7A2jVTCtZrfKimFq2II',
```

Comment:
This doc is shared within Google but not externally, so you'll need to either share this publicly or mock it.

This will work for mocking the endpoint:

```
const JSON = require("JSON");
const Promise = require("Promise");

const mockData = {
  spreadsheet_id: "abcdefghijklmnop",
  age: 1,
  address: "A",
  job: "it"
};

mock("sendHttpRequest", () => {
  return Promise.create((resolve) => {
    resolve({
      "statusCode": 200,
      "body": JSON.stringify({ "values": [["100"]]}),
      "headers": {
        "cache-control": "max-age=120"
      }
    });
  });
});

runCode(mockData).then((resp) => {
  assertThat(resp).isString();
  assertThat(resp).isEqualTo("100");
});
```

Reply:
Thank you so much for your advice.
Now I’m able to understand the mock api to override the behavior of Sandboxed APIs!!!
I removed the real spreadsheet id and I changed it to mock api.

Thanks!!